### DISCUSSION TITLE
Blueprint for Community Syncs

### DISCUSSION THREAD

**Discussion by errose28** [apache/ozone#6157] [Ideas]
At the last U.S. community sync, we had a discussion about how to run community syncs going forward. I'm opening this discussion to share ideas and hopefully construct a plan so that we can make the best use of this time. I will start with some ideas that were already shared out by various people.

## Document the Process

Whatever process we decide to use for community syncs, we should document it for reference. This could go under the `Community` or `Developer Guide` sections in the new website.

## Have a Rotating Community Sync Leader

We can gather a group of volunteers who would like to lead community syncs and rotate through them, so that each person does a week and we always know who the next community sync leader will be. Some sort of automation to notify the next week's community sync leader would be good to have. Maybe a GitHub action that automatically starts the discussion thread and tags the next leader?

## Standardize Meeting Plan and Notes in Github Discussions

It would be good to have a template that we use for all the community syncs. For example, the Tuesday before the meeting, a Github discussions thread is created. This thread is moderated by the community sync leader for that week. The inital agenda would have some of the same items every week (PRs pending review, go over notes from the other time zone's community sync, open forum for questions, etc), but people can comment in the discussion to add things they want to talk about. These could be design proposals or feature updates. If there are no volunteers the sync leader may recruit people to present as well.

## Better Coordination Between US and APAC Community Syncs

Currently the APAC community syncs are well documented by @chensammi. It would be good to standardize the format used among both community syncs so that people who do not attend can easily find the discussion notes for each week regardless of time zone. We can also take time in the syncs to go over the notes from the other time zone's sync.

Please add your ideas here and we can work on standardizing the process so that it is a beneficial time for everyone.

**Comment by errose28** [comment]
Looks like it's feasible to use github actions to auto create discussions and tag a rotating selection of leaders. [Here is the workflow on my fork](https://github.com/errose28/ozone/actions/runs/7895362311). At the bottom of the workflow output it linked to the [test discussion](https://github.com/errose28/ozone/discussions/3) that was created. I got an email notification when this discussion was created.

Here's how I imagine this working:
- A similar workflow is committed to the Ozone repo.
- A set of volunteers to rotate through as sync leaders are defined in the file. The list can be updated with a PR to add or remove people who would like to be in the roster.
- The workflow runs once a week on the Tuesday before the next community sync. It tags the leader and posts the template to start the agenda thread.
- We can edit the github bot's comment to change the leader if that person cannot make it. I have edit permissions on the bot's comment in my test.
- When starting the next week's sync thread, it will extract the leader of the previous sync from the last discussion and resume from the next spot in the list.

**Reply by kerneltime** [reply]
Sounds like a plan.

**Comment by umamaheswararao** [comment]
>A set of volunteers to rotate through as sync leaders are defined in the file. The list can be updated with a PR to add or remove people who would like to be in the roster.

I think we need to start a thread for collecting volunteers?

**Reply by errose28** [reply]
Sure. I have not had the bandwidth to drive this side project to completion yet, but we can get a mail thread started to get volunteers in the meantime. Probably good to get volunteers for both timezone syncs as well.
